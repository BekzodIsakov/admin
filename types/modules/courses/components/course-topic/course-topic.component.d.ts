/**
 * <CourseTopic />
 */
export const CourseTopic: any;
//# sourceMappingURL=course-topic.component.d.ts.map