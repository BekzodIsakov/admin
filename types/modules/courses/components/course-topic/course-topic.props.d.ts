export function useCourseTopicProps(_: any): {
    lesson: any;
};
//# sourceMappingURL=course-topic.props.d.ts.map