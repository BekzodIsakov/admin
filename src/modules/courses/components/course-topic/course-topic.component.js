import React from 'react';
import { hoc } from '@utils/hoc';
import { useCourseTopicProps } from './course-topic.props';
import {
  CourseTopicContainer,
  CourseTopicDescription
} from './course-topic.style';
import { Heading } from '@components/atoms';
import { Space } from '@components/space';

/**
 * <CourseTopic />
 */
const CourseTopic = hoc(useCourseTopicProps, ({ lesson }) => (
  <CourseTopicContainer>
    <Heading variant='h1' color='black'>
      {lesson?.title}
    </Heading>
    <Space space='12px' />
    <CourseTopicDescription
      dangerouslySetInnerHTML={{ __html: lesson?.body }}
      data-testid='course-topic'
    ></CourseTopicDescription>
  </CourseTopicContainer>
));

export { CourseTopic };
