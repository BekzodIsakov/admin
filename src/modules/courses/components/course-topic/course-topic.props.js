import { useActiveLessonSelector } from '@courses/store';
// import { useEffect } from 'react';
// import { useDispatch } from 'react-redux';

/**
 * <CourseTopic /> props
 */
export const useCourseTopicProps = _ => {
  // const dispatch = useDispatch();
  const lesson = useActiveLessonSelector();

  // useEffect(() => {
  //   if (!lesson) return;
  //   dispatch(
  //     coursesActions.fetchCourseHomeworks({
  //       course_id: lesson?.course_id,
  //       chapter_id: lesson?.chapter_id,
  //       lesson_id: lesson?.id
  //     })
  //   );
  // }, [dispatch, lesson]);

  return {
    lesson
  };
};
