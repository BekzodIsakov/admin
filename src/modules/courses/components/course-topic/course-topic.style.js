import styled from 'styled-components'

export const CourseTopicContainer = styled.div`

`

export const CourseTopicDescription = styled.div`
  padding-right: 20px;
  color: rgba(0,0,0,0.5);
`