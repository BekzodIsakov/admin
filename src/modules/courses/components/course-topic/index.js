export * from './course-topic.component';
