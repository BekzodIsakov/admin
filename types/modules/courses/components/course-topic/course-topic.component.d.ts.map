{"version":3,"file":"course-topic.component.d.ts","sourceRoot":"","sources":["../../../../../src/modules/courses/components/course-topic/course-topic.component.js"],"names":[],"mappings":"AAUA;;GAEG;AACH,8BAUG"}