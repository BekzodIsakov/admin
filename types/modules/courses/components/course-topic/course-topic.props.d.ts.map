{"version":3,"file":"course-topic.props.d.ts","sourceRoot":"","sources":["../../../../../src/modules/courses/components/course-topic/course-topic.props.js"],"names":[],"mappings":"AAOO;;EAkBN"}