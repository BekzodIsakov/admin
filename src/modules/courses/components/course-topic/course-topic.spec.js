import { customRender } from '@test-utils/helpers';
import { screen } from '@testing-library/react';

import { CourseTopic } from './course-topic.component';

const setup = ({ lessonTitle }) => {
  return customRender(<CourseTopic lessonTitle={lessonTitle} />);
};

test('Course topic', () => {
  const { asFragment } = setup({ lessonTitle: 'About + someText' });
  expect(asFragment).toMatchSnapshot();
});

test('course topic props', () => {
  setup({ lessonTitle: 'About + someText' });
  const courseTopic = screen.getByTestId('course-topic');
  expect(courseTopic).toBeTruthy();
});
