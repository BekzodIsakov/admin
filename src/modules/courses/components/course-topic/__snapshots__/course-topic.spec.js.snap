// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course topic 1`] = `[Function]`;
